contracts/PythOracle.sol

```diff
@@ -1,155 +1,129 @@
 
-    // Get the current price, confidence and metadata from the oracle.
-    // Please see https://docs.pyth.network/consumers/best-practices for best practices on consuming the data
-    // returned by this method.
-    function getPriceInfo() public view returns (PriceInfo memory){
-        PriceInfo memory info;
+        uint256 offset = 0;
 
-        // Read price account
-        bytes memory accData = QueryAccount.data(priceAccount, 208, 32);
+        // Skip: magic (4) + ver (4) + atype (4) + size (4) + ptype (4)
+        offset += 20;
 
-        info.price = readLittleEndianSigned64(accData.toUint64(0));
-        info.confidence = readLittleEndianUnsigned64(accData.toUint64(8));
-        info.status = PriceStatus(readLittleEndianUnsigned32(accData.toUint32(16)));
-        info.corporateAction = CorporateAction(readLittleEndianUnsigned32(accData.toUint32(20)));
-        info.updateSlot = readLittleEndianUnsigned64(accData.toUint64(24));
+        priceFeed.expo = readLittleEndianSigned32(data.toUint32(offset));
+        offset += 4;
 
-        return info;
-    }
+        priceFeed.maxNumPublishers = readLittleEndianUnsigned32(data.toUint32(offset));
+        offset += 4;
 
-    // Get the time-weighted average price (twap) and confidence (twac). Both of these values are
-    // the significands of fixed-point numbers, `twap * 10 ^ exponent`. Also returns the last slot
-    // when an update occurred.
-    function getTimeWeightedInfo() public view returns (int64 twap, uint64 twac, uint64 last_slot) {
-        bytes memory accData = QueryAccount.data(priceAccount, 32, 48);
-        last_slot = readLittleEndianUnsigned64(accData.toUint64(0));
-        twap = readLittleEndianSigned64(accData.toUint64(16));
-        twac = readLittleEndianUnsigned64(accData.toUint64(40));
-    }
+        priceFeed.numPublishers = readLittleEndianUnsigned32(data.toUint32(offset));
+        offset += 4;
 
-    // Get the price/confidence from the second-most-recent price update. This method returns
-    // a valid price from a time before the current price (as returned by getPriceInfo()).
-    // price and confidence are the significands of fixed-point numbers of the form `price * 10^exponent`.
-    function getPreviousUpdate() public view returns (int64 price, uint64 confidence, uint64 slot) {
-        bytes memory accData = QueryAccount.data(priceAccount, 176, 24);
-        slot = readLittleEndianUnsigned64(accData.toUint64(0));
-        price = readLittleEndianSigned64(accData.toUint64(8));
-        confidence = readLittleEndianUnsigned64(accData.toUint64(16));
-    }
+        // Skip: last_slot (8) + valid_slot (8)
+        offset += 16;
 
-    /////////////////////////////////////////////////////////////////////////////////////
-    // Chainlink adapter interface
-    // This interface replicates the chainlink Data Feeds API, excluding the historical API.
-    /////////////////////////////////////////////////////////////////////////////////////
```

Comment:
I suggest keeping the chainlink adapter interface. This is going to make it easier for EVM protocols to switch to Pyth when (if) they move to solana. In fact, it's probably good to have a chainlink adapter in our EVM sdk.

Feel free to do this in a separate PR though -- doesn't seem straightforward here because this method assumes the price feed id is provided to the constructor. I think there's probably a nice way to do this as a wrapper around `PythOracle`.